### PR TITLE
when changing or deleting a deploy group projects#show stage stayed u…

### PR DESCRIPTION
**Pull request by grosser**
…nchanged

@zendesk/runway 

### Risks
 - None

### CODE REVIEW

File: test/models/deploy_group_test.rb [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 describe DeployGroup do
-  let(:prod_env) { environments(:production) }
+  let(:stage) { stages(:test_staging) }
+  let(:environment) { environments(:production) }
+  let(:deploy_group) { deploy_groups(:pod1) }
+
+  def self.it_expires_stage(method)
+    it "expires stages when #{method}" do
+      stage.deploy_groups << deploy_group
+      stage.update_column(:updated_at, 1.minute.ago)
```
Comment (sandlerr) [inline test/models/deploy_group_test.rb]: can this be set in the `let` block that creates the fixture?

**Comment by sandlerr** [comment]
:+1:

File: test/models/deploy_group_test.rb (hunk 1 above)
Comment (grosser) [inline test/models/deploy_group_test.rb]: lets are cached per test case not test class, so there would be no benefit in using it.
A faster option would be to use timecop or make the fixture itself old, but that would add gem/mental overhead so ... meh ...

**Comment by grosser** [comment]
Ill take that +1 and run with it, let me know if you have an idea to make it better :)

File: test/models/deploy_group_test.rb (hunk 1 above)
Comment (sandlerr) [inline test/models/deploy_group_test.rb]: My suggestion was to make an old fixture - not sure what we can do here but in factorygirl you could go `let(:stage) { create(:stage, updated_at: 1.minute.ago) }` or similar

Comment (grosser) [inline test/models/deploy_group_test.rb]: create is even more expensive then an update_column :)

Comment (sandlerr) [inline test/models/deploy_group_test.rb]: is there not an implicit create going on here? If this all in memory then great!

Comment (grosser) [inline test/models/deploy_group_test.rb]: it's using a fixture -> no create in each test

**Pull request merged**